pkg/services/serviceaccounts/manager/service.go

```diff
@@ -48,6 +48,10 @@ func ProvideServiceAccountsService(
 	serviceaccountsAPI := api.NewServiceAccountsAPI(cfg, s, ac, routeRegister, s.store, permissionService)
 	serviceaccountsAPI.RegisterAPIEndpoints()
 
+	if err := s.store.UpdateAPIKeysExpiryDate(context.Background(), cfg.ServiceAccountExpirationDayLimit); err != nil {
```

Comment:
I think we need a better solution than to perform this on every restart of grafana and is thinking if we really care. Would it work to just ignore previously created token and only apply the restriction on new tokens?

Reply:
I believe that by not addressing existing tokens with an unlimited expiration date, we're not addressing the security vulnerability that this feature is trying to address.

This has been discussed in the design document. If you have a strong stance on this, I'd be happy to discuss this with the team.

Let me know what you think.